Release Changes to Main

t-bello7 opened the pull request:
This pull request contains the following changes to implement a working backend API for an hotel booking application
- Design and Develop a schema diagram
- Create authenticated controllers 
- Add Test for controllers, models and integrations
- Add documentation to API using rswag

[A link to the frontend pull request ](https://github.com/t-bello7/hotel-frontend/pull/21)

t-bello7 committed:
- add linter check

t-bello7 committed:
- add devise gem for authentication
- add cancan gem for authorization
- add rspec-swag for documentation
- create schema based on UML diagram

t-bello7 committed:
add roles column to users

t-bello7 committed:
fix linter issues

stephnna committed:
Merge pull request #9 from t-bello7/milestone-1

Milestone 1

stephnna committed:
generate routing and integration tests

stephnna committed:
instal rack-cors

stephnna committed:
Add user api

t-bello7 committed:
update integration tests

t-bello7 committed:
update integration tests

t-bello7 committed:
- fix linter issues

t-bello7 committed:
Merge pull request #10 from t-bello7/develop-api

Develop api

t-bello7 committed:
- refactor api to be v1
- deploy to fly

stephnna committed:
implement controllers

stephnna committed:
Fix linters

t-bello7 committed:
fix issues

t-bello7 committed:
change to v1

t-bello7 committed:
- fix merge issues:
:

t-bello7 committed:
add authentication endpoints

t-bello7 committed:
deploy

t-bello7 committed:
- fix linter issues and deploy

t-bello7 committed:
Merge pull request #11 from t-bello7/add-control

implement controllers

Meltrust commented on README.md:
```diff
@@ -68,6 +68,8 @@ After you're finished please remove all the comments and instructions!
 
 ## 🛠 Built With <a name="built-with"></a>
 
+### UML DIAGRAM
```
- [ ]  On the "License" section of your readme, the link is broken because it points to a file that does not exist in your project.

### Tips and extra help

Go [here](https://github.com/microverseinc/readme-template), grab the `MIT.md` file from that repo, paste it on the root of your project, and link to it in the "License" section of your readme.

- [x] In the same license section, the note `NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish.` is just a placeholder with instructions for you, so you can add a license.  Kindly remove it after you add your license.

- [x] There are no setup instructions about how to run your project properly.  Kindly add those.  Please, don't forget to include the required commands, like `bundle, rails db:create, reails db:migrate`, along with specific instructions about how to run your tests (e.g. `rake rswag`).

- [x] Most of your readme is only placeholders from the Microverse readme template.  Kindly add the content for all the required sections, with proper info about your project like features, description, etc. 

- [x] Kindly,  in the "authors" section, add the correct info on the authors, with the correct links to their social media profiles, so they can be contacted.

That way, your app will be well documented 💪

Meltrust replied:
- [ ]  You are using Rspec and Swagger, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

Meltrust replied:
- [ ] Your API has integrations (swagger) tests but is missing unit tests.  Kindly add some for your models and/or controllers in order to ensure your app's integrity. 💪

Meltrust reviewed (changes requested):
Hi @t-bello7,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Integration tests are passing✔️
-  API is working well✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

t-bello7 committed:
- create License

t-bello7 committed:
Create License

t-bello7 committed:
Merge branch 'add-control' into add-license-1

t-bello7 committed:
- add models unit test

t-bello7 committed:
- fix requests tests

t-bello7 committed:
Merge pull request #13 from t-bello7/add-license-1

Create License and Update README

stephnna committed:
Fix authorization, routes and models issues

stephnna committed:
Fix rubocop errors

t-bello7 committed:
- add active storage records
- update serializer, models and controllers of hotels to accept images

t-bello7 committed:
Merge pull request #14 from t-bello7/fix-authorization

Fix authorization, routes and models issues

t-bello7 committed:
Merge remote dev branch

t-bello7 committed:
- Merge branch 'dev' into add-license-1
- update serializer, models and controllers of rooms to accept images

stephnna committed:
Merge pull request #15 from t-bello7/handle-image

Add Implementation for handling images

stephnna committed:
I updated the sagger documentation

stephnna committed:
Fix rubocop errors

t-bello7 committed:
Merge pull request #16 from t-bello7/update-swagger

Updated the swagger documentation

t-bello7 committed:
- allow booking routes to be accessible without userId
- create bookings by current user
- add username, role, and eamil to token authentication

stephnna committed:
Fix jwt issue

stephnna committed:
Fix rubocop and login issue for new users

stephnna committed:
Update Readme file

stephnna committed:
Merge pull request #17 from t-bello7/update-routes-controllers

Remove Nested routes for bookings

Meltrust commented on README.md:
````diff
@@ -97,137 +55,98 @@ After you're finished please remove all the comments and instructions!
-
 ```sh
-  bin/rails test test/models/article_test.rb
+  rspec spec 
````
- [ ]  When I run `rspec spec` according to your instructions, I get this:

![image](https://user-images.githubusercontent.com/57421823/216636137-5dac6078-b7e7-4cd5-bd2d-7425b652bda3.png)

If these tests are passing on your end and there are special procedures to run your tests properly without failing, kindly add those instructions to your readme.  👍

Meltrust reviewed (changes requested):
Hi @t-bello7,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------
